src/sds_data_model/raster.py

```diff
@@ -55,11 +55,71 @@
 #     masks: Generator[MaskTile, None, None]
 #     metadata: Metadata
 
-#     def to_data_arrays(self) -> TiledDataArrayLayer:
-#         data_arrays = (mask.to_data_array(name=self.name) for mask in self.masks)
+#    def to_data_arrays(self) -> TiledDataArrayLayer:
+#        data_arrays = (mask.to_data_array(name=self.name) for mask in self.masks)
 
 #         return TiledDataArrayLayer(
 #             name=self.name,
 #             data_arrays=data_arrays,
 #             metadata=self.metadata,
 #         )
+
+
+# temporary func to read raster + resample & reshape if necessary
+from pathlib import Path
+from typing import List, Literal, Optional, Sequence, Union
+
+from rasterio.drivers import raster_driver_extensions
+from xarray import Dataset, open_dataset
+
+from sds_data_model._raster import _resample_and_reshape
+from sds_data_model._vector import _check_layer_projection
+
+
+def read_dataset_from_file(
+    data_path: str,
+    band: Union[List[int], List[str], None] = None,
+    categorical: Union[bool, Sequence[bool]] = False,
+    nodata: Optional[float] = None,
+    engine: Optional[str] = None,
+    decode_coords: Union[bool, None, Literal["coordinates", "all"]] = "all",
```

Comment:
Use `Optional` for a `Union` that could be `None`:

```suggestion
    data_path: str,
    band: Optional[List[int], List[str]] = None,
    categorical: Union[bool, Sequence[bool]] = False,
    nodata: Optional[float] = None,
    engine: Optional[str] = None,
    decode_coords: Optional[Union[bool, Literal["coordinates", "all"]]] = "all",
```